cpp/src/arrow/util/cpu-info.cc

```diff
@@ -132,6 +135,43 @@ bool RetrieveCacheSize(int64_t* cache_sizes) {
   }
   return true;
 }
+
+bool RetrieveCPUInfo(int64_t& hardware_flags, std::string& model_name) {
```

Comment:
Could you change this function signature to use pointers (e.g. `std::string* model_name`) instead of mutable references?